src/objects/zcl_abapgit_object_intf.clas.abap

```diff
@@ -401,24 +401,19 @@ CLASS zcl_abapgit_object_intf IMPLEMENTATION.
 
 
   METHOD zif_abapgit_object~deserialize.
-
-    DATA: ls_vseointerf TYPE vseointerf.
-
-    io_xml->read( EXPORTING iv_name = 'VSEOINTERF'
-                  CHANGING cg_data = ls_vseointerf ).
+    DATA: lt_source     TYPE rswsourcet,
+          ls_clskey     TYPE seoclskey.
 
     IF iv_step = zif_abapgit_object=>gc_step_id-abap.
-
-      IF ls_vseointerf-clsproxy = abap_true.
-        " Proxy interfaces are managed via SPRX
-        deserialize_proxy( iv_transport ).
-        RETURN.
-      ENDIF.
-
-      deserialize_abap( ii_xml     = io_xml
-                        iv_package = iv_package ).
-
-      deserialize_docu( io_xml ).
+      deserialize_xml( ii_xml       = io_xml
+                       iv_package   = iv_package
+                       iv_transport = iv_transport ).
+
+      ls_clskey-clsname = ms_item-obj_name.
+      lt_source = zif_abapgit_object~mo_files->read_abap( ).
+      mi_object_oriented_object_fct->deserialize_source(
+        is_key    = ls_clskey
+        it_source = lt_source ).
```

Comment:
semantics here are changed?

proxy objects will now try to deserialize source?

docu is handled before source instead of after? I dont know if this is significant, but we want to keep this code working

Reply:
there are some changes regarding PRIVATE vs PROTECTED, which are good changes. But this deserialization is critical functionality, and just makes it more difficult to review.
Perhaps do a PR with safe easily reviewable changes, or do multiple PRs?

Reply:
> proxy objects will now try to deserialize source?
You are right. Missed that. This should be double-checked.

> docu is handled before source instead of after? I dont know if this is significant, but we want to keep this code working
I think this should be fine because documentation is completely independent of the source. Also, I don't expect descriptions overwritten by updating the source code. 
Are automated tests available for updating interfaces with descriptions?

Reply:
integration tests = https://ci.abapgit.org but personally I do not have the infrastructure to execute it

Reply:
@larshp You were right. The order of updating the description is relevant. One integration test fails, currently.

Reason: When a component (attribute, method, ...) is added, the object handler for interfaces adds an entry into the text table, but not in the corresponding header tables. When the source code and its metadata is updated (later), the attribute is unknown for Class Builder's functionality. Therefore, the text is not retrieved and assigned to the new component. 

This means, since we only update the text table, the descriptions at the end.